### DISCUSSION TITLE
How to prevent close alert dialog with esc?

### DISCUSSION THREAD

**Discussion by igortullio** [radix-ui/primitives#1434] [Help] [answered]
## Question

I need the alert not to close when I click Esc.

**Comment by BiswaViraj** [comment, marked as answer]
https://www.radix-ui.com/docs/primitives/components/alert-dialog#content
Pass `onEscapeKeyDown` props to the `Content` as
`onEscapeKeyDown={(event) => {
   event.preventDefault();
  }
}`

**Reply by Manbearpixel** [reply]
Great solution 🙌

I enhanced this a little further to allow the escape key to still work *unless* the element that had focus when escape was pressed (i.e., an input element) had this attribute `data-prevent-modal-close-on-escape="true"`

```
<input data-prevent-modal-close-on-escape="true" />
```

Followed by the prop for `Dialog.Content`

```
onEscapeKeyDown={(e) => {
  if ((e.target as HTMLElement).dataset.preventModalCloseOnEscape) {
    e.preventDefault();
  }
}}
```
